### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by Yidnekachew-cmd**
### In this branch, we have added the following features;

- [ ] A home page showing a list of items.
     - When the page loads, the web app retrieves data from: 
         - The selected API shows the list of items on the screen.
         - The Involvement API to show the item likes.
- [ ] A popup window with more data about an item that we can use to comment on it.
     - When the popup loads, the webapp retrieves data from:
         -  The selected API shows details about the selected item.
         -  The Involvement API to show the item comments.
- [ ] First, you find API sand and the API  allows us to:
     - Get a list of items with a unique item id (or generate the unique id).
     - For a given item, get detailed information about it.
- [ ] We use [Meals DB](https://www.themealdb.com/api.php): data about meals.
- [ ] We use  [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes and comments).
- [ ] We followed the layout of the wireframes provided.
- [ ] When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- [ ] When the user clicks on the "Comments" button, the Comments popup appears.
- [ ] Home page header and navigation similar to the given mockup.
- [ ] Home page footer similar to the given mockup.
- [ ] We have counters in all the interfaces that show:
   - The number of items (home).
   - The number of comments (comments popup).
- [ ] Set up the repository on GitHub and use Gitflow.
- [ ] Set up webpack.
- [ ] Set up a JavaScript testing library (Jest).
- [ ] We added a video with a maximum length of 5 minutes with a description of the project and a demo of the project features.
- [ ] We added the video link in the `readme` file and talk about the project.
- [ ] In this project we also practice giving code reviews to each other.

### CODE REVIEW

File: modules/commentCtr.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const comments = async (count) => {
+  const h4 = document.getElementById('comments');
+  h4.innerHTML = `Comments(${count})`;
+};
```
Comment (amejid) [inline modules/commentCtr.js:4]: - [ ] Great job on the UI and all the functionalities. :+1: Let's work on the counter functions to achieve the project requirement. Currently, this function is getting some number and displaying it on the screen. I think the counting is done in the other file `getComments.js`.
 
Screenshot from 2023-03-22 15-13-24

I believe the counting should be done inside the counter function and displaying the count should be in the other file, that way it'll make it easier to test the counters.

**Possible solution**

`commentCtr.js`

Screenshot from 2023-03-22 15-27-19

`getComments.js`

Screenshot from 2023-03-22 15-27-39

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @Yidnekachew-cmd and @fuadmoin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- Functionalities working as expected :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- renaming the folder from `Src` to `src` for webpack to work properly.

Screenshot from 2023-03-22 14-53-26


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: __tests__/commentCtr.test.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import comments from '../modules/commentCtr.js';
+
+describe('Check counter all the comments', () => {
+  test('Should be show Comments(29) ', () => {
+    document.body.innerHTML = '<h4 id=\'comments\'></h4>';
+    const comment = document.getElementById('comments');
+    comments(29);
+    expect(comment.textContent).toEqual('Comments(29)');
+  });
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply __tests__/commentCtr.test.js]: Fixed.

File: __tests__/foodCounter.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+describe('food counter', () => {
+  test('length of the array should be zero', () => {
+    const zeroArr = [];
+    expect(foodCounter(zeroArr)).toBe(0);
+  });
+  test('length of the array should be three', () => {
+    const threeArr = [1, 2, 3];
+    expect(foodCounter(threeArr)).toBe(3);
+  });
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply __tests__/foodCounter.test.js]: Fixed.

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @Yidnekachew-cmd  @fuadmoin   , 

Great job for implemented  the some of the suggested changes  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No Linters error
✔️ You used correct Git-Flow.
✔️ Good commit messages
✔️ Professional READEME 


almost

## Required Changes ♻️
_Check the comments below the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @fuadmoin, and @Yidnekachew-cmd👋, 

### You did a marvelous job on the Javascript group capstone project and this is one of the best projects I have seen so far!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## To Highlight 🎉
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- All tests are passed. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**